tst/EnergyPlus/unit/ThermalChimney.unit.cc

```diff
@@ -0,0 +1,1148 @@
+        "    Verbose,                 !- Internal Variable Availability Dictionary Reporting",
+        "    Verbose;                 !- EMS Runtime Language Debug Output Level",
+
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    using namespace EnergyPlus::DataIPShortCuts;
+
+    lNumericFieldBlanks.allocate(1000);
+    lAlphaFieldBlanks.allocate(1000);
+    cAlphaFieldNames.allocate(1000);
+    cNumericFieldNames.allocate(1000);
+    cAlphaArgs.allocate(1000);
+    rNumericArgs.allocate(1000);
+    lNumericFieldBlanks = false;
+    lAlphaFieldBlanks = false;
+    cAlphaFieldNames = " ";
+    cNumericFieldNames = " ";
+    cAlphaArgs = " ";
+    rNumericArgs = 0.0;
+
+    bool ErrorsFound = false;
+    // Read objects
+    HeatBalanceManager::GetProjectControlData(outputFiles(), ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+    HeatBalanceManager::GetZoneData(ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+    HeatBalanceManager::GetWindowGlassSpectralData(ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+    HeatBalanceManager::GetMaterialData(outputFiles(), ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+    HeatBalanceManager::GetConstructData(ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+    SurfaceGeometry::GetGeometryParameters(outputFiles(), ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+
+    SurfaceGeometry::CosBldgRotAppGonly = 1.0;
+    SurfaceGeometry::SinBldgRotAppGonly = 0.0;
+    SurfaceGeometry::GetSurfaceData(outputFiles(), ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+    ScheduleManager::ProcessScheduleInput(OutputFiles::getSingleton());
+    ScheduleManager::ScheduleInputProcessed = true;
+
+    DataHeatBalance::Zone(2).HasWindow = true;
+    DataHeatBalance::Zone(4).HasWindow = true;
+    EnergyPlus::DataHeatBalSurface::TempSurfIn.allocate(TotSurfaces);
+    DataHeatBalance::HConvIn.allocate(TotSurfaces);
+    DataHeatBalance::HConvIn = 0.1;
+    DataHeatBalSurface::TempSurfIn = 25.00;
+    DataHeatBalSurface::TempSurfIn(8) = 25.92;
+    DataHeatBalSurface::TempSurfIn(9) = 25.92;
+    DataHeatBalSurface::TempSurfIn(12) = 26.99;
+    DataHeatBalSurface::TempSurfIn(24) = 22.99;
+    DataHeatBalFanSys::MAT.allocate(DataGlobals::NumOfZones);
+    DataHeatBalFanSys::ZoneAirHumRat.allocate(DataGlobals::NumOfZones);
+    DataHeatBalFanSys::MAT = 23.0;
+    DataHeatBalFanSys::ZoneAirHumRat = 0.01;
+    DataEnvironment::OutBaroPress = 101325.0;
+    DataHeatBalFanSys::MCPThermChim.allocate(DataGlobals::NumOfZones);
+    DataHeatBalFanSys::ThermChimAMFL.allocate(DataGlobals::NumOfZones);
+    DataHeatBalFanSys::MCPTThermChim.allocate(DataGlobals::NumOfZones);
+    ScheduleManager::Schedule(1).CurrentValue = 1.0;
+    DataHeatBalance::ZnAirRpt.allocate(DataGlobals::NumOfZones);
+    // No EMS
+    ThermalChimney::GetThermalChimney(ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+    //ThermalChimney::CalcThermalChimney();
+    //EXPECT_NEAR(ThermalChimney::ThermalChimneyReport(1).OverallTCVolumeFlow, 0.015668, 0.0001);
```

Comment:
Why can't you call the thermal chimney model now with EMSOverrideOn = false ?

Reply:
@rraustad It is a good question. There was a unit test failure from Linux testing at 172507808-Issue7044 (lgu1234) - x86_64-Linux-Ubuntu-18.04-gcc-7.5: OK (2856 of 2857 tests passed, 0 test warnings)  

Here is a link to the failure.
https://nrel.github.io/EnergyPlusBuildResults/EnergyPlus-350cb0cd7c35338e5b2006aab4ed1c09abe75f40-x86_64-Linux-Ubuntu-18.04-gcc-7.5.html

When I run the unit test from my local computer (VS2017), no failure occurs. In addition, other CI machines also do not show any unit test failure. That was why I commented the test to see what happened. 

I can call the the thermal chimney model now with EMSOverrideOn = false, but the Linux machine got the different answer. Since I don't have Linux, I don't know how to fix it. In addition, this issue is to create an EMS actuator, the unit test shows that the EMS actuator works. This is the purpose for this unit test. I can create a new issue to investigate Linux machine performance.

Reply:
@lgu1234 I pulled locally on Linux and updated with latest develop.  I re-enabled these lines you had commented.  I was able to build/run your new unit test both by itself as well as in conjunction with the full unit test suite.  I pushed the changes up to see if Linux will be happy this time.  If not, I'll concede for the moment and pull it back out.

Reply:
@Myoldmopar Great. Thanks for doing it.